### ISSUE TITLE
web browser accessibility failure

### ISSUE THREAD

**Issue by githubgnulinux** [tootsuite/mastodon#13383]
Please consider w3c accessibility guidance in web site design. Unable to view web site content after sign-in

TypeError: document.scrollingElement is undefined

t/<@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:583359
t/<@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:581740
y@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:158098
k/<@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:158466
k@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:158423
N</n.componentDidMount@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:584676
Du@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1057479
t.unstable_runWithPriority@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1077060
Xa@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1001987
Fu@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1053829
wu@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1044148
Ja/<@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1002278
t.unstable_runWithPriority@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1077060
Xa@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1001987
Ja@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1002223
$a@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1002158
gu@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1040714
Co.enqueueSetState@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:1006939
_.prototype.setState@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:951563
t/</<@https://mastodon.social/packs/js/common-ee9226a17a0221aba66c.js:2:274838

getScrollTop (webpack:///app/javascript/mastodon/components/scrollable_list.js:154:4)
throttle( (webpack:///app/javascript/mastodon/components/scrollable_list.js:54:24)
invokeFunc (webpack:///node_modules/lodash/debounce.js:97:13)
leadingEdge (webpack:///node_modules/lodash/debounce.js:107:21)
leadingEdge (webpack:///node_modules/lodash/debounce.js:101:2)
componentDidMount (webpack:///app/javascript/mastodon/components/scrollable_list.js:142:4)
fd (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:6324:54)
unstable_runWithPriority (webpack:///node_modules/scheduler/cjs/scheduler.production.min.js:270:11)
fg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:2794:9)
Yj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:6105:2)
Zj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:5408:6)
fg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:2829:16)
unstable_runWithPriority (webpack:///node_modules/scheduler/cjs/scheduler.production.min.js:270:11)
fg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:2794:9)
fg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:2824:6)
jg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:2814:2)
Kj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:5191:91)
enqueueSetState (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:3091:4)
setState (webpack:///node_modules/react/cjs/react.production.min.js:60:2)
this.setState (webpack:///app/javascript/mastodon/features/ui/components/bundle.js:79:8)

**Comment by ThibG** [comment]
This has nothing to do with accessibility, and is a duplicate of an issue you already opened (#13266), which should be fixed (but maybe not deployed on mastodon.social).

Also, whenever reporting a front-end issue, please tell which browser you are using.

**Comment by githubgnulinux** [comment]
Excuse, but misunderstood. Is accessibility about a web site being designed to conform to w3 standards, then people being able to access the web site with the requirement that they use a web browser that conforms to that standard?

With reference to issue 13266, if it is fixed, but not deployed on mastodon.social, where to go to test the fix please?

**Comment by ThibG** [comment]
In that context, “accessibility” refers to designing apps or websites so that they can be properly used by people with various disabilities (by making it usable with various assistive technologies).

This particular issue has nothing to do with that and is a compatibility issue with old browsers (you still haven't told which browser you use). The set of standards defined by the W3C is, thankfully, not set in stone, and there are various evolving standards. In this case, that API is part of a W3C Working Draft (so, not a proper standard, but something aiming to be one), and has been widely implemented in browsers for years.

I am fairly sure this is deployed on mastodon.social, now. If the problem continues to occur, try clearing your cache, and if it keeps happening, please post an up-to-date stacktrace as well as information on what browser you use.

**Comment by githubgnulinux** [comment]
Have used various web browsers (webkit e.g. midori is completely inaccessible to mastodon; gnu/linux firefox 60 quantum is successful; gnu/linux firefox 75 is inaccessible; gnu/linux chromium 70 inaccessible).

If mastodon is designed to comply with html5, users are asked to review browser compatibility to w3 validator, or html5test.com, web browser version becomes less important, compared to conformance with web standards

**Comment by ProgVal** [comment]
@githubgnulinux Like ThibG said, your issue is with compatibility, not accessibility. Compatibility is about working for all browsers; accessibility is about working for all users.

Also, the w3c validator is for checking HTML conformance of a website, not which Javascript APIs are used. And html5test.com is for checking what web standards your browser supports, and is of no importance to website developers.

This doesn't mean this bug won't be fixed; both are important to mastodon developers. And from what I understand, this bug is already fixed in the latest version of Mastodon (which is not yet deployed everywhere, but then you should tell your instance about this).

**Comment by ThibG** [comment]
@githubgnulinux I concur with everything ProgVal said, and I will add that I just tested Mastodon with Midori 7, and it works fine (even without the fix for #13266). I can't see why it would crash (assuming that's what you mean by “inaccessible” given your previous comments) with Firefox 75 or Chromium 70, if it does, it's another issue altogether, that needs reporting with a corresponding stack trace.

**Comment by githubgnulinux** [comment]
Sent: Thursday, April 09, 2020 at 8:58 PM
From: "Valentin Lorentz" <notifications@github.com>
 
> @githubgnulinux[https://github.com/githubgnulinux] Like ThibG said, your issue is with compatibility, not accessibility. Compatibility is about working for all browsers; accessibility is about working for all users.

Personal opinion, that these are inter-related but thanks for the definitions

> Also, the w3c validator is for checking HTML conformance of a website, not which Javascript APIs are used. And html5test.com is for checking what web standards your browser supports, and is of no importance to website developers.

This is very surprising to read: web site programmers do not care about web standards? Wow! Combustion engine engineers want to know the standard fuels; they don't invest in engine manufacture without consideration of the fuel standards available to their customers.

**Comment by ProgVal** [comment]
Sure it's useful for them to know the features supported by the browser their use personally; but it's useless to check what other browsers support.

**Comment by githubgnulinux** [comment]
Sent: Thursday, April 16, 2020 at 9:13 PM
From: "Valentin Lorentz" <notifications@github.com>
 
> Sure it's useful for them to know the features supported by the browser their use personally; but it's useless to check what other browsers support.

With respect, "empathetic programming" could be considered as an appreciation of what _other_ people have available to use. It is impossible to make everyone happy, but (if you want) to maximise...

**Comment by ProgVal** [comment]
I don't understand what this has to do with html5test.com
